File: Python/Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        dic = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000}
        for index, ch in enumerate(s[:-1]):
            if dic[ch] < dic[s[index + 1]]:
                count -= dic[ch]
            else:
                count += dic[ch]
        return count + dic[s[-1]]
        
        

File: cpp/Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char, int> values;
        values['I'] = 1;
        values['V'] = 5;
        values['X'] = 10;
        values['L'] = 50;
        values['C'] = 100;
        values['D'] = 500;
        values['M'] = 1000;

        int num = 0;

        for (int i = 0; i < s.size() - 1; ++i) {
            if (values[s[i]] < values[s[i + 1]]) {
                num -= values[s[i]];
            } else {
                num += values[s[i]];
            }
        }
        return num + values[s[s.size() - 1]];
    }
};
